ereolen_support/ereolen_support.helper.inc

```diff
@@ -0,0 +1,279 @@
+          }
+          else {
+            $data['fields']['customfield_' . $jira_field_id] = check_plain($form_state['values'][$key]);
+          }
+        }
+      }
+    }
+  }
+
+  $requestParts = _ereolen_support_get_request_parts('POST', $data);
+
+  $result = drupal_http_request($requestParts['jira_path'] . '/issue', $requestParts['options']);
+  _ereolen_support_log_error($result);
+}
+
+/**
+ * Create the parts required to make a drupal_http_request.
+ *
+ * @param string $method_type
+ *   The request method.
+ * @param array $data
+ *   An optional data array.
+ *
+ * @return array
+ *   An array containing base path for the request and the options related to
+ *   the request.
+ *
+ * @return array
+ */
+function _ereolen_support_get_request_parts($method_type, $data = []) {
+  $jiraUsername = variable_get('jira_username');
+  $jiraPassword = variable_get('jira_password');
+  // Jira requires base64_encode auth.
+  $encoded_auth = base64_encode($jiraUsername . ':' . $jiraPassword);
+  $jira_api_path = variable_get('jira_domain') . '/rest/api/latest';
+
+  $options = [
+    'method' => $method_type,
+    'timeout' => 15, // Seconds.
+    'headers' => [
+      'Content-Type' => 'application/json',
+      'Authorization' => 'Basic '. $encoded_auth
+    ],
+  ];
+
+  // Add data variable to POST and PUT requests.
+  if ($method_type == 'POST' || $method_type == 'PUT' ) {
+    $options['data'] = json_encode($data);
+  }
+
+  $requestParts = [
+    'jira_path' => $jira_api_path,
+    'options' => $options
+  ];
+
+  return $requestParts;
+}
+
+/**
+ * Prepare an array for use in a select list.
+ *
+ * @param string $vocab_machine_name
+ *  A vocabulary machine name.
+ * @return array
+ *   A list of term names.
+ */
+function _ereolen_support_create_select_list($vocab_machine_name) {
+  $vacabulary = taxonomy_vocabulary_machine_name_load($vocab_machine_name);
+  $terms = taxonomy_get_tree($vacabulary->vid);
+  if ($terms) {
+    $list = [];
+    foreach ($terms as $term) {
+      $list[$term->name] = str_repeat('-', $term->depth) . $term->name;
+    }
+
+    return $list;
+  }
+  return [];
+}
+
+/**
+ * Set array keys as values for use in select list.
+ *
+ * @param $array
+ *  An array of values.
+ * @return array
+ *   An array with keys like values.
+ */
+function _ereolen_support_array_as_select($array) {
+  $list = [];
+  foreach ($array as $value) {
+    $list[$value] = $value;
+  }
+
+  return $list;
+}
+/**
+ * Provide a support ID grabbed from Drupal user.
+ *
+ * @return string|null
+ */
+function _ereolen_support_get_support_id() {
+  if (user_is_logged_in() && isset($_SESSION['ding_user']['user']['data'])) {
+    $user_data = $_SESSION['ding_user']['user']['data'];
+    if (isset($user_data['support_id'])) {
+      return $user_data['support_id'];
+    }
+  }
+  return NULL;
+}
+
+/**
+ * Add an entry to the eReolen support error log.
+ * We use a separate error log for easier access to potential errors. This also
```

Comment:
Missing empty line between shot and long docblock function comments.

Reply:
Added.